src/d/actor/d_a_obj_eskban.cpp

```diff
@@ -3,94 +3,421 @@
+    J3DModelData* model_data =
+        static_cast<J3DModelData*>(dComIfG_getObjectRes(M_arcname, ESKBAN_BDL_ESKBAN));
+    JUT_ASSERT(261, model_data != 0);
+    mpModel = mDoExt_J3DModel__create(model_data, 0, 0x11020203U);
+    M_smoke = new dPa_smokeEcallBack();
+    JUT_ASSERT(264, M_smoke != 0);
+    return mpModel != NULL;
 }
 
 /* 000001FC-00000368       .text Create__Q211daObjEskban5Act_cFv */
-s32 daObjEskban::Act_c::Create() {
-    /* Nonmatching */
+int daObjEskban::Act_c::Create() {
+    this->cullMtx = mpModel->getBaseTRMtx();
+    init_mtx();
+    fopAcM_setCullSizeBox(this, -500, -1, -500, 500, 500, 500);
+    mCheckStts.Init(0xff, 0xff, this);
+    mCheckCyl.Set(cyl_check_src);
+    mCheckCyl.SetC(this->current.pos);
+    mCheckCyl.SetStts(&mCheckStts);
+
+    mCameraStts.Init(0xff, 0xff, this);
+    mCameraCyl.Set(cyl_camera_src);
+    cXyz offset(300, 450, 300);
+    mCameraCyl.SetC(this->current.pos + offset);
+    mCameraCyl.SetStts(&mCameraStts);
+
+    mCheckSphStts.Init(0xff, 0xff, this);
+    mCheckSph.Set(sph_check_src);
+    offset.set(0, 400, 0);
+    mCheckSph.SetC(this->current.pos + offset);
+    mCheckSph.SetStts(&mCheckSphStts);
+
+    mActorState = 0;
+    mRemainingSmokeAnm = 0;
+    mIsVisible = 1;
+    return TRUE;
 }
 
 /* 000003A4-000004D0       .text Mthd_Create__Q211daObjEskban5Act_cFv */
-void daObjEskban::Act_c::Mthd_Create() {
-    /* Nonmatching */
+s32 daObjEskban::Act_c::Mthd_Create() {
+    s32 phase_state;
+    fopAcM_SetupActor(this, Act_c);
+    M_smoke = NULL;
+
+    s32 swSave = param_get_swSave();
+    if (fopAcM_isSwitch(this, swSave)) {
+        return cPhs_UNK3_e;
+    }
+    phase_state = dComIfG_resLoad(&mPhs, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, ESKBAN_DZB_ESKBAN, NULL, 0x1020);
+        JUT_ASSERT(336, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+    }
+    return phase_state;
 }
 
+daObjEskban::Act_c::Act_c() {}
+
 /* 000009C0-00000A10       .text Delete__Q211daObjEskban5Act_cFv */
 BOOL daObjEskban::Act_c::Delete() {
-    /* Nonmatching */
+    if (this->M_smoke) {
+        M_smoke->end();
+        M_smoke = NULL;
+    }
+    return TRUE;
 }
 
 /* 00000A10-00000A68       .text Mthd_Delete__Q211daObjEskban5Act_cFv */
-void daObjEskban::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+BOOL daObjEskban::Act_c::Mthd_Delete() {
+    s32 result = MoveBGDelete();
+    if (fpcM_CreateResult(this) != cPhs_UNK3_e) {
+        dComIfG_resDelete(&mPhs, M_arcname);
+    }
+    return result;
 }
 
 /* 00000A68-00000B00       .text set_mtx__Q211daObjEskban5Act_cFv */
 void daObjEskban::Act_c::set_mtx() {
-    /* Nonmatching */
+    this->shape_angle = this->current.angle;
+    mDoMtx_stack_c::transS(this->current.pos);
+    mDoMtx_stack_c::ZXYrotM(this->shape_angle);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    mDoMtx_copy(mDoMtx_stack_c::get(), this->M_tmp_mtx);
```

Comment:
Remove unnecessary this->

Reply:
I removed it for `M_tmp_mtx` and everything that is an obvious class member

For the snake_case unprefixed variables inherited from `fop_Ac_c`, it feels useful to not confuse them with local variables, but I can also remove those if you prefer